src/TestFramework/Microsoft.Rest.ClientRuntime.Azure.TestFramework/ConnectionString.cs

```diff
@@ -0,0 +1,337 @@
+﻿namespace Microsoft.Rest.ClientRuntime.Azure.TestFramework
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Reflection;
+    using System.Text;
+    using System.Text.RegularExpressions;
+
+    /// <summary>
+    /// This class represents the connection string being set by the user
+    /// e.g. TEST_CSM_ORGID_AUTHENTICATION="AADTenant=72f98AAD-86f1-2d7cd011db47;ServicePrincipal=72f98AAD-86f1-2d7cd011db47;Password=tzT2+LJBRkSAursui7/Qgo+hyQQ=;SubscriptionId=5562fbd2-HHHH-WWWW-a55d-lkjsldkjf;BaseUri=https://management.azure.com/;AADAuthEndpoint=https://login.windows.net/;GraphUri=https://graph.windows.net/"
+    /// </summary>
+    public class ConnectionString
+    {
+        #region fields
+        Dictionary<string, string> _keyValuePairs;
```

Comment:
please explicitly put the private keyword. Since we switch between so many different languages it is better to have access modifier than remember the default one.